petab/lint.py

```diff
@@ -65,6 +93,79 @@ def condition_table_is_parameter_free(condition_df):
+    check_parameter_df(problem.parameter_df)
+    parameterId_is_string(problem.parameter_df)
+    parameterScale_is_valid(problem.parameter_df)
+    parameterBounds_are_numeric(problem.parameter_df)
+    parameterEstimate_is_boolean(problem.parameter_df)
+    parameterId_is_unique(problem.parameter_df)
+    check_parameterBounds(problem.parameter_df)
+    return True
+
+def parameterId_is_string(parameter_df):
+    """Check if all entries in the parameterId column of the parameter table are string and not empty"""
+
+    for parameterId in parameter_df['parameterId']:
+        if isinstance(parameterId, str):
+            if parameterId[0].isdigit():
+                raise ValueError('parameterId ' + parameterId + ' starts with integer')
+        else:
+            raise ValueError('Empty parameterId found')
+
+    return True
+
+def parameterId_is_unique(parameter_df):
+    """Check if the parameterId column of the parameter table is unique"""
+    if len(parameter_df['parameterId']) == len(set(parameter_df['parameterId'])):
+        return True
+    else:
+        raise AssertionError('parameterId column in parameter table is not unique')
+
+
+def parameterScale_is_valid(parameter_df):
+    """Check if all entries in the parameterScale column of the parameter table are
+    'lin' for linear, 'log' for natural logarithm or 'log10' for base 10 logarithm """
+
+    for parameterScale in parameter_df['parameterScale']:
+        if parameterScale not in ['lin', 'log', 'log10']:
+            raise AssertionError('Expected "lin", "log" or "log10" but got "'+parameterScale+'"')
+
+    return True
+
+def parameterBounds_are_numeric(parameter_df):
+    """Check if all entries in the lowerBound and upperBound column of the parameter table are
+    numeric """
+
+    for lowBound in parameter_df['lowerBound']:
+        try:
+            float(lowBound)
+        except:
+            raise AssertionError(f'Expected float but got {lowBound, type(lowBound)} as lower bound')
+
+    for upBound in parameter_df['upperBound']:
+        try:
+            float(upBound)
+        except:
+            raise AssertionError(f'Expected float but got {upBound, type(upBound)} as upper bound')
+
+    return True
+
+def check_parameterBounds(parameter_df):
+    """Check if all entries in the lowerBound are smaller than upperBound column in the parameter table"""
+    for element in range(len(parameter_df['lowerBound'])):
+        if not parameter_df['lowerBound'][element] <= parameter_df['upperBound'][element]:
+            raise AssertionError(f'lowerbound larger than upperBound in {parameter_df["parameterId"][element]}')
+
+
+def parameterEstimate_is_boolean(parameter_df):
+    """Check if all entries in the estimate column of the parameter table are boolean """
+
+    for estimate in parameter_df['estimate']:
+        if not int(estimate) in [True, False]:
+            raise AssertionError(f'Expected boolean but got {estimate, type(estimate)} in estimate')
```

Comment:
We agreed to have estimate as 1/0, right? Then "Expected boolean" might be confusing. If we agreed to have estimate as "True/False", I would explicitly write that in the error message, i.e. "Expected boolean True/False".